test/node/content/BatchResponseContent.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { assert } from "chai";
+
+import { BatchResponseContent } from "../../../src/content/BatchResponseContent";
+
+const redirect = {
+	id: "1",
+	status: 302,
+	headers: {
+		location: "https://b0mpua-by3301.files.1drv.com/y23vmagahszhxzlcvhasdhasghasodfi",
+	},
+};
+const forbidden = {
+	id: "3",
+	status: 401,
+	body: {
+		error: {
+			code: "Forbidden",
+			message: "...",
+		},
+	},
+};
+const okWithBody = {
+	id: "2",
+	status: 200,
+	body: {
+		"@odata.context": "https://graph.microsoft.com/v1.0/$metadata#Collection(microsoft.graph.plannerTask)",
+		value: [],
+	},
+};
+const okWithoutBody = {
+	id: "4",
+	status: 204,
+	body: null,
+};
+
+describe("BatchResponseContent.ts", () => {
+	describe("update", function() {
+		this.timeout(20 * 1000);
+
+		it("Should update instance with nextLink responses", (done) => {
+			const responses = [redirect, forbidden];
+			const batchRes = new BatchResponseContent({
+				responses,
+				"@odata.nextLink": "dummy nextlink",
+			});
+			batchRes.update({
+				responses: [okWithoutBody, okWithBody],
+			});
+			assert.isDefined(batchRes.getResponseById("2"));
```

Comment:
I suggest that we test that the BatchResponseContent.Update doesn't delete/alter the existing batch responses. Might as well:

```javascript
assert.isDefined(batchRes.getResponseById("1"));
assert.isDefined(batchRes.getResponseById("2"));
assert.isDefined(batchRes.getResponseById("3"));
assert.isDefined(batchRes.getResponseById("4"));
```

This does lead me to wonder why `update()` is public. I don't think we want want customers adding responses to BatchResponseContent.

Reply:
Done